openpathsampling/engines/dynamics_engine.py

```diff
@@ -656,10 +669,21 @@ def iter_generate(self, initial, running=None, direction=+1,
         if final_error is not None:
             yield trajectory
             logger.info("Through frame: %d", len(trajectory))
+            self._clear_snapshot_cache(snapshot)
             raise final_error
 
         logger.info("Finished trajectory, length: %d", len(trajectory))
         yield trajectory
+        self._clear_snapshot_cache(snapshot)
+
+    def _clear_snapshot_cache(self, snapshot):
+        if self.clear_snapshot_cache and snapshot is not None:
+            try:
+                # isolate the AttributeError to just getting this method
+                clear_cache = snapshot.clear_cache
+            except AttributeError:
+                clear_cache = lambda: None
```

Comment:
This can be solved slightly better (in my opinion)
```suggestion
            clear_cache = getattr(snapshot, 'clear_cache', lambda: None)
```

Reply:
Yup. And that also fixes the uncovered line in the local tests 😉